Dockerfile

```diff
@@ -1,24 +1,28 @@
+# Stage 1: Precompile TestRunner and Module Cache
 FROM swift:6.1.2 AS builder
-# WORKDIR /opt/testrunner
-
-COPY src/TestRunner ./
-
-# Print Installed Swift Version
 RUN swift --version
-#RUN swift package clean
+
+# Build TestRunner executable
+WORKDIR /TestRunner
+COPY src/TestRunner .
 RUN swift build --configuration release
 
+# Build WarmUp package
+# Build directory and final working paths should be equal for reuse of ModuleCache.
+WORKDIR /opt/test-runner
```

Comment:
Not intuitive, but it will actually throw an error if path to .pcm will be different.